### PR TITLE
add insecure-registry checkbox to modify imagestream

### PR DESCRIPTION
**Pull request by aweiteka**
Initial support for ["insecure registry" boolean](https://docs.openshift.org/latest/rest_api/openshift_v1.html#v1-tagimportpolicy), applied per tag.

### CODE REVIEW

**Comment by petervo** [comment]
The test in pkg/kubernetes/tests/test-tags.html needs to be updated.

File: pkg/kubernetes/scripts/tags.js [hunk 1]
```diff
@@ -73,8 +73,10 @@
         });
         var tags = [ ];
         angular.forEach(names, function(name) {
-            if (name in already)
+            if (name in already) {
+                already[name].importPolicy = { "insecure": insecure };
                 tags.push(already[name]);
+            }
             else
                 tags.push({ name: name });
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/tags.js]: Why does this only apply to tags that already exist? What about newly created tags here on line 81?

File: pkg/kubernetes/tests/test-tags.html [hunk 2]
```diff
@@ -66,13 +66,13 @@ <h2 id="qunit-userAgent"></h2>
...
-            spec = data.buildSpec(["2.5", "latest", "second"], spec);
+            spec = data.buildSpec(["2.5", "latest", "second"], spec, true);
             deepEqual(spec, {
                 "dockerImageRepository": "busybox",
                 "tags": [
                     { "name": "2.5" },
                     { "annotations": null, "from": { "kind": "DockerImage", "name": "busybox:latest" },
-                        "generation": 2, "importPolicy": {}, "name": "latest" },
+                        "generation": 2, "importPolicy": { "insecure": true }, "name": "latest" },
                     { "name": "second" }
```
Comment (stefwalter) [inline pkg/kubernetes/tests/test-tags.html]: This line should have insecure as well.

File: pkg/kubernetes/tests/test-tags.html [hunk 3]
```diff
@@ -66,13 +66,13 @@ <h2 id="qunit-userAgent"></h2>
         "imageTagData",
         function(data) {
             var spec = angular.extend({ }, specData);
-            spec = data.buildSpec(["2.5", "latest", "second"], spec);
+            spec = data.buildSpec(["2.5", "latest", "second"], spec, true);
             deepEqual(spec, {
                 "dockerImageRepository": "busybox",
                 "tags": [
                     { "name": "2.5" },
```
Comment (stefwalter) [inline pkg/kubernetes/tests/test-tags.html]: This line should have insecure as well.

File: pkg/kubernetes/scripts/tags.js [hunk 4]
```diff
@@ -64,7 +64,7 @@
         return names;
     }
 
-    function buildSpec(names, spec) {
+    function buildSpec(names, spec, insecure) {
```
Comment (stefwalter) [inline pkg/kubernetes/scripts/tags.js]: Can we make this so if insecure is null or undefined the relevant importPolicy is not set or reset?

Comment (aweiteka) [inline pkg/kubernetes/scripts/tags.js]: With this there are now two tag-specific use cases that I see for this dialog:

1. adding or removing a tag
1. controlling if those tags are insecure

A registry is either insecure or secure so my assumption is that it applies to all images. I'm not sure we can support both use cases above without also passing in the state of the insecure flag.

Comment (stefwalter) [inline pkg/kubernetes/scripts/tags.js]: Good point. I agree. Nhe fix below still needs to be made for new tags.

Comment (stefwalter) [inline pkg/kubernetes/scripts/tags.js]: Good point. I agree. The fix below still needs to be made for new tags.

Comment (stefwalter) [inline pkg/kubernetes/scripts/tags.js]: Good point. I agree. The fix below still needs to be made for new tags.

**Pull request closed without merging**